R/plot.R

```diff
@@ -573,27 +498,66 @@ plot_predictions <- function(data = NULL,
       )
   }
 
-  plot <- plot +
-    ylab("True and predicted values")
-
-  # facet if specified by the user
-  if (!is.null(facet_formula)) {
-    if (facet_wrap_or_grid == "facet_wrap") {
-      plot <- plot +
-        facet_wrap(facet_formula, scales = scales, ncol = ncol)
-    } else {
-      plot <- plot +
-        facet_grid(facet_formula, scales = scales)
-    }
-  }
-
   return(plot)
 }
 
 
 
+#' @title Filter Data Used for Plotting
+#'
+#' @description
+#' Filter data before it gets passed to [plot_predictions()] using arbitrary
+#' filter statements. In contrast to the usual behaviour of filtering functions,
+#' rows do net get removed, but instead corresponding values of either
+#' 'true_value' or 'predction' will be replaced with `NA`. This allows to filter
+#' truth and prediction data independently (either turning 'true_value' or
+#' 'prediction' or both into `NA`).
+#'
+#' @inheritParams score
+#' @param what character vector that determines which values should be turned
+#' into `NA`. If `what = "truth"`, values in the column 'true_value' will be
+#' turned into `NA`. If `what = "forecast"`, values in the column 'prediction'
+#' will be turned into `NA`. If `what = "both"`, values in both column will be
+#' turned into `NA`.
+#' @param ... logical statements used to filter the data
+#' @return A data.table
+#' @importFrom rlang enexprs
+#' @keywords plotting
+#' @export
+#'
+#' @examples
+#' filter_data (
+#'     example_continuous,
+#'     what = "truth",
+#'     target_end_date <= "2021-07-22",
+#'     target_end_date > "2021-05-01"
+#'   )
+
```

Comment:
Personally I think I would have a filter_forecast and filter_truth set of functions with an internal doing the actual filtering.

In the case of the filter_truth function you could also use this to specify the unique combinations rather than in plot_predictions